src/main/java/com/dataloom/hazelcast/serializers/SearchResultStreamSerializer.java

```diff
@@ -0,0 +1,47 @@
+package com.dataloom.hazelcast.serializers;
+
+import java.io.IOException;
+import java.util.List;
+import java.util.Map;
+
+import org.springframework.stereotype.Component;
+
+import com.dataloom.hazelcast.StreamSerializerTypeIds;
+import com.dataloom.search.requests.SearchResult;
+import com.hazelcast.nio.ObjectDataInput;
+import com.hazelcast.nio.ObjectDataOutput;
+import com.kryptnostic.rhizome.pods.hazelcast.SelfRegisteringStreamSerializer;
+
+@Component
+public class SearchResultStreamSerializer implements SelfRegisteringStreamSerializer<SearchResult> {
+
+    @Override
+    public void write( ObjectDataOutput out, SearchResult object ) throws IOException {
+        out.writeLong( object.getNumHits() );
+        out.writeObject( object.getHits() );
+    }
+
+    @Override
+    public SearchResult read( ObjectDataInput in ) throws IOException {
+        long numHits = in.readLong();
+        List<Map<String, Object>> hits = in.readObject();
```

Comment:
In cases like where we know most of this stuff safely serialized as JSON, we can use the SmileMapper to generate an efficient binary representation. Alternatively, you could also use Kryo as well, but that is more complicated to setup due to thread safety issues.  https://jira.thedataloom.com/browse/BACKEND-832